submissions/evgeniy24/object-oriented-frogger-game/app.js

```diff
@@ -0,0 +1,101 @@
+
+const FIELD_WIDTH = 505,
+        START_POS_X = 202,
+        START_POS_Y = 404;  
+        CELL_WIDTH = 101; 
+        CELL_HEIGHT = 83;
+
+
+// Enemies our player must avoid
+const Enemy = function(x, y, speed, player) {
+    // Variables applied to each of our instances go here,
+    // we've provided one for you to get started
+    this.x = x;
+    this.y = y;
+    this.speed = speed;
+    // The image/sprite for our enemies, this uses
+    // a helper we've provided to easily load images
+    this.sprite = 'images/enemy-bug.png';
+    this.player = player;
+};
+
+// Update the enemy's position, required method for game
+// Parameter: dt, a time delta between ticks
+Enemy.prototype.update = function(dt) {
+    // You should multiply any movement by the dt parameter
+    // which will ensure the game runs at the same speed for
+    // all computers.
+    this.x += this.speed * dt;
+    if (this.x > FIELD_WIDTH) {
+         this.x = 0;
+         this.speed =  Math.floor(Math.random() * 256) + 20;
```

Comment:
Is 20 a minimal speed and 256 is a max speed before thresholding?
Just trying to guess.
Things like FIELD_WIDTH, CELL_HEIGHT are pretty talking.

Reply:
Yes, please see new commit changes. I defined new variables!